core/services/keystore/keys/terrakey/export.go

```diff
@@ -2,53 +2,44 @@ package terrakey
 
 import (
 	"encoding/hex"
-	"encoding/json"
 
-	keystore "github.com/ethereum/go-ethereum/accounts/keystore"
-	"github.com/pkg/errors"
+	"github.com/ethereum/go-ethereum/accounts/keystore"
+	keys_export "github.com/smartcontractkit/chainlink/core/services/keystore/keys"
 	"github.com/smartcontractkit/chainlink/core/utils"
 )
 
 const keyTypeIdentifier = "Terra"
 
 // FromEncryptedJSON gets key from json and password
 func FromEncryptedJSON(keyJSON []byte, password string) (Key, error) {
-	var export EncryptedTerraKeyExport
-	if err := json.Unmarshal(keyJSON, &export); err != nil {
-		return Key{}, err
-	}
-	privKey, err := keystore.DecryptDataV3(export.Crypto, adulteratedPassword(password))
-	if err != nil {
-		return Key{}, errors.Wrap(err, "failed to decrypt Terra key")
-	}
-	key := Raw(privKey).Key()
-	return key, nil
-}
-
-// EncryptedTerraKeyExport represents the Terra encrypted key
-type EncryptedTerraKeyExport struct {
-	KeyType   string              `json:"keyType"`
-	PublicKey string              `json:"publicKey"`
-	Crypto    keystore.CryptoJSON `json:"crypto"`
+	return keys_export.FromEncryptedJSON[keys_export.EncryptedKeyExport](
+		keyTypeIdentifier,
+		keyJSON,
+		password,
+		adulteratedPassword,
+		func(_ keys_export.EncryptedKeyExport, rawPrivKey []byte) (Key, error) {
+			return Raw(rawPrivKey).Key(), nil
+		},
+	)
 }
 
 // ToEncryptedJSON returns encrypted JSON representing key
 func (key Key) ToEncryptedJSON(password string, scryptParams utils.ScryptParams) (export []byte, err error) {
-	cryptoJSON, err := keystore.EncryptDataV3(
+	return keys_export.ToEncryptedJSON[keys_export.EncryptedKeyExport](
```

Comment:
Can these type params be inferred?
```suggestion
	return keys_export.ToEncryptedJSON(
```

Reply:
ah good point, forgot to clean that part up when I changed a parameter to include this type